# Array of location
Location[] location